src/Clock.tsx

```diff
@@ -0,0 +1,68 @@
+import { Component } from 'react';
+
+type State = {
+  currentTime: string;
+};
+type Props = {
+  name: string;
+};
+
+function toFormattedTime(date: Date): string {
+  return date.toUTCString().slice(-12, -4);
+}
+
+export class Clock extends Component<Props, State> {
+  state: Readonly<State> = {
+    currentTime: toFormattedTime(new Date()),
+  };
+
+  intervalId = 0;
+
+  componentDidMount(): void {
+    this.intervalId = window.setInterval(this.handleUpdateTime, 1000);
+  }
+
+  componentDidUpdate(prevProps: Readonly<Props>): void {
+    const previousName = prevProps.name;
+    const currentName = this.props.name;
+
+    if (previousName !== currentName) {
+      // eslint-disable-next-line no-console
+      console.debug(`Renamed from ${prevProps.name} to ${this.props.name}`);
```

Comment:
if you want to prevent linter error just use window
```suggestion
      // eslint-disable-next-line no-console
      window.console.debug(`Renamed from ${prevProps.name} to ${this.props.name}`);
```